src/panels/lovelace/cards/hui-history-graph-card.ts

```diff
@@ -177,11 +172,10 @@ export class HuiHistoryGraphCard extends LitElement implements LovelaceCard {
 
     if (
       changedProps.has("_config") &&
-      (!this._subscribed ||
-        oldConfig?.entities !== this._config.entities ||
-        oldConfig?.hours_to_show !== this._hoursToShow)
+      (oldConfig?.entities !== this._config.entities ||
+        oldConfig?.hours_to_show !== this._config.hours_to_show)
     ) {
-      this._unsubscribeHistoryTimeWindow();
+      await this._unsubscribeHistoryTimeWindow();
```

Comment:
Wait for unsubscribe before resubscribe

Reply:
You probably already noticed, but we have the same pattern here:

```typescript
src/dialogs/more-info/ha-more-info-history.ts:    this._unsubscribeHistoryTimeWindow();
src/dialogs/more-info/ha-more-info-history.ts:  private _unsubscribeHistoryTimeWindow() {
src/dialogs/more-info/ha-more-info-history.ts:      this._unsubscribeHistoryTimeWindow();
src/panels/lovelace/cards/hui-history-graph-card.ts:    this._unsubscribeHistoryTimeWindow();
src/panels/lovelace/cards/hui-history-graph-card.ts:  private _unsubscribeHistoryTimeWindow() {
src/panels/lovelace/cards/hui-history-graph-card.ts:      this._unsubscribeHistoryTimeWindow();
src/panels/lovelace/cards/hui-map-card.ts:    this._unsubscribeHistoryTimeWindow();
src/panels/lovelace/cards/hui-map-card.ts:  private _unsubscribeHistoryTimeWindow() {
src/panels/lovelace/cards/hui-map-card.ts:        this._unsubscribeHistoryTimeWindow();
src/panels/lovelace/cards/hui-map-card.ts:      this._unsubscribeHistoryTimeWindow();
src/panels/lovelace/header-footer/hui-graph-header-footer.ts:    this._unsubscribeHistoryTimeWindow();
src/panels/lovelace/header-footer/hui-graph-header-footer.ts:  private _unsubscribeHistoryTimeWindow() {
src/panels/lovelace/header-footer/hui-graph-header-footer.ts:      this._unsubscribeHistoryTimeWindow();
```

And a similar pattern in logbook  ./panels/logbook/ha-logbook.ts

```typescript
    
  private _unsubscribe(): void {
    if (this._subscribed) {
      this._subscribed.then((unsub) =>
        unsub
          ? unsub().catch(() => {
              // The backend will cancel the subscription if
              // we subscribe to entities that will all be
              // filtered away
            })
          : undefined
      );
      this._subscribed = undefined;
    }
  }
    
```

Reply:
I can go back and adjust the other ones once this PR is merged and we have an a agreed upon pattern